plugins/woocommerce-blocks/assets/js/blocks/product-collection/constants.ts

```diff
@@ -80,7 +80,7 @@ export const DEFAULT_ATTRIBUTES: Pick<
 		columns: 3,
 		shrinkColumns: true,
 	},
-	queryContextIncludes: [ 'collection' ],
+	queryContextIncludes: [ 'collection', 'selectedReference' ],
```

Comment:
> I think we could keep it in query in the same way as we included filterable

I believe that "filterable" is also a part of the "query context". I'm simplifying this and thinking about how the "query" should only have "query-related" arguments, as described by the WordPress REST documentation, along with some specific query arguments for products.

It seems that everything else related to handling queries by external factors such as location, filters, and collections is essentially considered as "query context." To make this clearer, think about the way you parse the `filterable` key in PHP - it's within the block's context, just like location. I don't have a strong opinion, honestly, and I think this restrucure should happen on a different PR.

So, my approach would be to first consolidate the location and selected reference **here** and initiate a discussion about the "putting-all-together" soon.

Reply:
@xristos3490  Are you suggesting that in a different PR, we should create a new attribute called `queryContext` which will contain all the context related to query i.e. `location`, `selectedReference` etc.? (Name doesn't have to be `queryContext`, it can be anything. I am just trying to understand the concept here)

Reply:
@kmajinak I don't have a strong opinion here. I can move it to "query", but I will wait until we make a decision on this 🙂

Reply:
> Are you suggesting that in a different PR, we should create a new attribute called queryContext which will contain all the context related to query i.e. location, selectedReference etc.? (Name doesn't have to be queryContext, it can be anything. I am just trying to understand the concept here)

I am just thinking of the `$request` params instead of something like:
```
- woocommerceHandpickedProducts: Array
- ...[other query args]...
- location: Array
- queryContextIncludes: Array {
    - collection
    - selectedReference
}
```

To (ideally) something way simpler like:
```
- woocommerceHandpickedProducts: Array
- ...[other query args]...
- queryContext: Array {
    - collection
    - *currentReference
}
```

Or a bit less destructive from now:
```
- woocommerceHandpickedProducts: Array
- ...[other query args]...
- *currentReference: Array
- queryContextIncludes: Array {
    - collection
}
```
* In both examples, `currentReference` is the merged product of the `location` data and the `selectedReference` attribute.

Reply:
> In both examples, currentReference is the merged product of the location data and the selectedReference attribute.

I like the name `currentReference`. I will make the changes. 

CC: @kmanijak Please let me know in case you don't agree with this name 🙂

> I personally prefer approach (1) because it helps to maintain clarity between what is considered a "query arg" and woo-related context data. However, I understand that queryContextIncludes is designed to work with attributes, and location is not exactly one. If that seems challenging, option (2) is a good alternative for now.

I can make the changes as per approach (1), if we think that it's a good long term solution. I like this approach as it combines all woo query context in single variable i.e.
- Everything from `queryContextIncludes`
- `collection`
- `filterable`
- `currentReference`

@kmanijak What do you think about this? If you agree, I can try making the changes in this PR.

Reply:
My idea was the same as an approach 2 from Chris' suggestions. I think it'd be good to discuss it quickly on a sync call 🙌

Reply:
Got it. Let's have a quick sync call to discuss this :handshake: